rust/db_handling/src/interface_signer.rs

```diff
@@ -219,9 +219,51 @@ pub fn backup_prep (database_name: &str, seed_name: &str) -> Result<String, Erro
+            };
+            let collision_display = format!("\"base58\":\"{}\",\"path\":\"{}\",\"has_pwd\":{},\"identicon\":\"{}\",\"seed_name\":\"{}\"", address_base58, address_details.path, address_details.has_pwd, hex_identicon, seed_name);
+            Ok(format!("\"seed_name\":\"{}\",\"network_title\":\"{}\",\"network_logo\":\"{}\",\"network_specs_key\":\"{}\",\"suggested_derivation\":\"{}\",\"collision\":{{{}}}", seed_name, network_specs.title, network_specs.logo, hex::encode(network_specs_key.key()), suggest, collision_display))
+        },
+        None => Ok(format!("\"seed_name\":\"{}\",\"network_title\":\"{}\",\"network_logo\":\"{}\",\"network_specs_key\":\"{}\",\"suggested_derivation\":\"{}\"", seed_name, network_specs.title, network_specs.logo, hex::encode(network_specs_key.key()), suggest)),
+    }
+}
+
+/// Function to show (dynamically) if the derivation with the provided path and no password already exists
+/// and if it exists, prints its details
+pub fn dynamic_path_check (database_name: &str, seed_name: &str, path: &str, network_specs_key_hex: &str) -> String {
+    let content = match NetworkSpecsKey::from_hex(network_specs_key_hex) {
+        Ok(network_specs_key) => {
+            match get_network_specs(database_name, &network_specs_key) {
+                Ok(network_specs) => {
+                    match derivation_check (seed_name, path, &network_specs_key, database_name) {
+                        Ok(DerivationCheck::BadFormat) => String::from("\"button_good\":false"),
+                        Ok(DerivationCheck::Password) => String::from("\"button_good\":true,\"where_to\":\"pwd\""),
+                        Ok(DerivationCheck::NoPassword(None)) => String::from("\"button_good\":true,\"where_to\":\"pin\""),
+                        Ok(DerivationCheck::NoPassword(Some((multisigner, address_details)))) => {
+                            let address_base58 = print_multisigner_as_base58(&multisigner, Some(network_specs.base58prefix));
+                            let hex_identicon = match make_identicon_from_multisigner(&multisigner) {
+                                Ok(a) => hex::encode(a),
+                                Err(_) => String::new(),
+                            };
+                            let collision_display = format!("\"base58\":\"{}\",\"path\":\"{}\",\"has_pwd\":{},\"identicon\":\"{}\",\"seed_name\":\"{}\"", address_base58, address_details.path, address_details.has_pwd, hex_identicon, seed_name);
+                            format!("\"button_good\":false,\"collision\":{{{}}}", collision_display)
+                        },
+                        Err(e) => format!("\"error\":\"{}\"", <Signer>::show(&e)),
+                    }
+                },
+                Err(e) => format!("\"error\":\"{}\"", <Signer>::show(&e)),
+            }
+        },
+        Err(e) => format!("\"error\":\"{}\"", <Signer>::show(&e)),
+    };
+    format!("{{\"derivation_check\":{{{}}}}}", content)
```

Comment:
is this actually json? or some format to bind to kotlin or something?

Reply:
This is actual JSON.